pkgs/build-support/node/fetch-yarn-deps/yarn-berry-config-hook.sh

```diff
@@ -0,0 +1,30 @@
+yarnBerryConfigHook(){
+    echo "Executing yarnBerryConfigHook"
+
+    # Use a constant HOME directory
+    mkdir -p /tmp/home
+    export HOME=/tmp/home
```

Comment:
Sorry if I missed this in another comment, but why is this necessary? Would `$NIX_BUILD_TOP`/`$TMP` not work here?

Reply:
> Sorry if I missed this in another comment, but why is this necessary? Would `$NIX_BUILD_TOP`/`$TMP` not work here?

Hmm I'm not sure that `NIX_BUILD_TOP` is a good idea, see https://github.com/NixOS/nixpkgs/issues/189691 . And the idea here I think is to use a constant and known temporary directory (not using `mktemp -d`). I don't remember exactly what were the issues I had when I added that in [`a84f3ca` (#318015)](https://github.com/NixOS/nixpkgs/pull/318015/commits/a84f3ca3d824b306ab3080b11495327763d7f2f4) , but I'm open to debate this. Is $TMP an environment variable that is constant for every derivation build?

Reply:
Should be constant on Linux, but not necessarily on darwin because darwin doesn't have linux's filesystem namespaces / chroot, so there's no way to ensure the same build dir is free.

I'm not sure about the nix shell issue though. It feels like a bug in nix shell tbh. It should set these env vars appropriately, but I get that might be difficult to fix right now, so happy to keep it /tmp if this is an issue.

I'd still leave a comment linking to the github issue though.

Reply:
According to https://nix.dev/manual/nix/stable/language/derivations `TMP` should be OK too. Would you be satisfied if instead of `/tmp/home` would be replaced by `$TMP/home`?

Reply:
Done

Reply:
> Done

@gador while at it, could you do me a favor and add another commit before the commit titled `fetch-yarn-deps: add yarn-berry support` that will modify the `yarnConfigHook` as well to use `$TMP/home`?

Reply:
sure